pkg/networkserver/grpc_gsns.go

```diff
@@ -781,11 +781,15 @@ func (ns *NetworkServer) handleUplink(ctx context.Context, up *ttnpb.UplinkMessa
 
 // newDevAddr generates a DevAddr for specified EndDevice.
 func (ns *NetworkServer) newDevAddr(context.Context, *ttnpb.EndDevice) types.DevAddr {
-	// TODO: Allow generating DevAddr from a prefix. (https://github.com/TheThingsNetwork/lorawan-stack/issues/130)
+	// TODO: Add tests
 	nwkAddr := make([]byte, types.NwkAddrLength(ns.netID))
 	random.Read(nwkAddr)
 	nwkAddr[0] &= 0xff >> (8 - types.NwkAddrBits(ns.netID)%8)
+
+	prefix := ns.devAddrPrefixes[random.Intn(len(ns.devAddrPrefixes))]
```

Comment:
We need to check configuration in `pkg/networkserver.New()` that there are in fact device address prefixes.

Or, if there are none, add one empty based on the NetID.